### PR TITLE
fix(910B): fix bugs in 910B for varlen and fixlen FA

### PR DESCRIPTION
**Pull request by li126com**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
[issue 307](https://github.com/InternLM/InternEvo/issues/307)

## Modification
1. rotary_embedding为确保正确性采用gpu torch的版本。
2. 修改了unpack data情况下seq_len的计算逻辑。
3. 判定并修正了910B上all_reduce的op操作，由AVG转为SUM。

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by sunpengsdu** [comment]
补充PR的description

File: internlm/core/scheduler/pipeline_scheduler.py [hunk 1]
```diff
@@ -309,13 +309,14 @@ def _forward_step(
                 output_obj = loss_reduced
 
         moe_loss = (
-            sum(moe_losses) * gpc.config.loss.moe_loss_coeff
+            sum(moe_losses) * gpc.config.loss.moe_loss_coeff  # pylint: disable=E0606
```
Comment (sunpengsdu) [inline internlm/core/scheduler/pipeline_scheduler.py:312]: why disable=E0606

File: internlm/model/ops/rotary_emb.py [hunk 2]
```diff
@@ -143,13 +143,14 @@ def rotary_emb_in_rotate_half_style(
...
     """
+    assert False, "This function has some bugs. You should not arrive here."
     # reformat cos/sin shape.
     cos = torch.cat((cos, cos), dim=-1)[None, :, None, :]
     sin = torch.cat((sin, sin), dim=-1)[None, :, None, :]
 
     if len(x.shape) == 5:
         q, k, _ = x.unbind(dim=2)
-
+        q, k = q.squeeze(dim=2), k.squeeze(dim=2)
```
Comment (sunpengsdu) [inline internlm/model/ops/rotary_emb.py:153]: 这不是在https://github.com/InternLM/InternEvo/pull/296/files 去掉了吗，为啥这里又改回来

File: internlm/core/scheduler/pipeline_scheduler.py (hunk 1 above)
Comment (li126com) [reply internlm/core/scheduler/pipeline_scheduler.py:312]: 这个有点怪就是在910B上执行pre-commit检查他会在这行出这个问题，只好disable了

**Comment by li126com** [comment]
> add description

done

File: internlm/model/ops/rotary_emb.py (hunk 2 above)
Comment (li126com) [reply internlm/model/ops/rotary_emb.py:153]: 没有，这个是我们当时说的他不是有个漏加的地方么，就是296的PR去掉了attention那块的squeeze，当时不是说embedding这应该加一个squeeze再

**Review by sunpengsdu — APPROVED** [review]
